docs/dev/style.md

```diff
@@ -93,3 +93,18 @@ later in the same file.
 Using consistent wording across Cirq is important for lowering users
 cognitive load. For rule governing naming, see the 
 [nomenclature guidelines](nomenclature.md).
+
+## Datetimes
+
+Prefer using timezone-aware `datetime` objects.
+
+```python
+import datetime
+dt = datetime.datetime.now(tz=datetime.timezone.utc)
+```
+
+Protobuf APIs will return "aware" `datetime` objects. JSON de-serialization will
+promote values to "aware" `datetime` objects upon deserialization. 
```

Comment:
Might also want to suggest that in cirq functions when we compare `datetime` instances (for equality or ordering) we should compare `.timestamp()` instead of comparing the datetime instances directly, since this will work for aware and naive datetimes just like serialization.

Reply:
wellllllllllllllll in https://github.com/quantumlib/Cirq/issues/5301#issuecomment-1119981947 (specifically the linked comment) the proposal was to compare the datetimes. This was in the context of dataclass-like member variables where the motivation was to mirror the behavior of the components. I suppose the tradeoffs could be different in functions where we just want it to work.

Propose adding:

> Comparing (or testing equality) between "naive" and "aware" `datetime` objects throws
an exception.
If you are implementing a class that has `datetime` member variables, delegate equality
and comparison operators to the built-in `datetime` equality and comparison operators.
If you're writing a function that compares `datetime` objects, you can defensively promote
them to "aware" objects or use their `.timestamp()` properties for a comparison that will
never throw an exception.

Reply:
SGTM